xbmc/windowing/X11/WinSystemX11.cpp

```diff
@@ -20,8 +20,6 @@
 
 #include "system.h"
 
-#if defined(HAVE_X11)
```

Comment:
The build system is weak in this regard because it includes files if it finds packages instead of what it really should include. @fetzerch works on a solution. In the meantime you could add another hack to this weird logic there:
https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/CMakeLists.txt#L15